src/shims/os_str.rs

```diff
@@ -215,35 +215,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         size: u64,
     ) -> InterpResult<'tcx, (bool, u64)> {
         let this = self.eval_context_mut();
-        let os_str = convert_path_separator(path.as_os_str(), &this.tcx.sess.target.target.target_os);
+        let os_str = convert_path_separator(path.as_os_str(), &this.tcx.sess.target.target.target_os, true);
         this.write_os_str_to_wide_str(&os_str, scalar, size)
     }
 }
 
 /// Perform path separator conversion if needed.
+/// if direction == true, Convert from 'host' to 'target'.
+/// if direction == false, Convert from 'target' to 'host'.
```

Comment:
Please, don't use magic booleans. On the call site it is impossible to say what they mean.
Introduce an `enum` with appropriately named variants instead.